lib/base_repository.rb

```diff
@@ -0,0 +1,27 @@
+class BaseRepository
```

Comment:
it's not something that you have to change 
but you could also create `BaseConcept` in `app/concepts/base` this should be more clear for other developers every concepts related class are in `concepts` directory

Reply:
I based my code on this:
https://github.com/monterail/mirror-consono-backend/blob/main/lib/base_repository.rb

Is that not considered best practices anymore? 🤔

Reply:
You mean the directory? It depends how generally you treat `concepts` directory and there is no good answer.

- If `concepts` directory is considered as a directory separated for things separated in terms of business meaning, then I think that `base` shouldn't be here and in that way your current solution is good.
- If `concepts` directory is considered as a directory separated by our understanding of scopes in-app (cause this is also one of the approaches) then `base` might be here and inside might be the base repository.

It always depends on global understanding of `concepts` directory and there is always more than one good answer.

Reply:
Wow that's a lot of knowledge! 📚 🚀 Thank you!